### ISSUE TITLE
make failed on ubuntu 16.04

### ISSUE THREAD

**Issue by onzone** [OpenShot/libopenshot#692]
I am getting this error while make in ubuntu..
```
[ 90%] Linking CXX executable openshot-player
../src/libopenshot.so.0.2.5: undefined reference to `avresample_alloc_context'
../src/libopenshot.so.0.2.5: undefined reference to `avresample_free'
../src/libopenshot.so.0.2.5: undefined reference to `avresample_close'
../src/libopenshot.so.0.2.5: undefined reference to `avresample_convert'
../src/libopenshot.so.0.2.5: undefined reference to `avresample_open'
collect2: error: ld returned 1 exit status
examples/CMakeFiles/openshot-player.dir/build.make:128: recipe for target 'examples/openshot-player' failed
make[2]: *** [examples/openshot-player] Error 1
CMakeFiles/Makefile2:983: recipe for target 'examples/CMakeFiles/openshot-player.dir/all' failed
make[1]: *** [examples/CMakeFiles/openshot-player.dir/all] Error 2
Makefile:165: recipe for target 'all' failed
make: *** [all] Error 2
```
Here is my CMake output:
```
-----------------------------------------------------------------
          Welcome to the OpenShot Build System!

CMake will now check libopenshot's build dependencies and inform
you of any missing files or other issues.

For more information, please visit <http://www.openshot.org/>.
-----------------------------------------------------------------

Generating build files for OpenShot with CMake 3.20.2
  Building libopenshot (version 0.2.5)
  SO/API/ABI Version: 19

-- Could NOT find ImageMagick (missing: ImageMagick_Magick++_LIBRARY ImageMagick_MagickCore_LIBRARY) (found version "6.8.9-9")
-- Looking for system jsoncpp
-- Could NOT find Resvg, using Qt SVG parsing instead (missing: Resvg_LIBRARIES Resvg_INCLUDE_DIRS) 
-- PYTHON_MODULE_PATH: /usr/local/lib/python3.6/dist-packages
-- Could NOT find Ruby (missing: Ruby_EXECUTABLE Ruby_INCLUDE_DIR Ruby_LIBRARY) (Required is at least version "1.8.0")
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Build configuration:
-- The following OPTIONAL packages have been found:

 * PkgConfig
 * PythonInterp (required version >= 3)
 * PythonLibs (required version >= 3)

-- The following REQUIRED packages have been found:

 * OpenShotAudio (required version >= 0.2.0)
 * jsoncpp
 * Qt5Widgets
 * FFmpeg
 * OpenMP
 * ZeroMQ
 * Qt5Core (required version >= 5.5.1)
 * Qt5Gui
 * Qt5
 * SWIG (required version >= 3.0)

-- The following features have been disabled:

 * Coverage, analyze test coverage and generate report
 * IWYU (include-what-you-use), Scan all source files with 'iwyu'
 * OpenCV algorithms, Use OpenCV algorithms
 * Unit tests, Compile unit tests for library functions

-- The following OPTIONAL packages have not been found:

 * ImageMagick
 * cppzmq
 * Resvg
 * Ruby
 * Doxygen
 * Catch2

-- Configuring done
WARNING: Target "pyopenshot" requests linking to directory "/usr/lib/python3.6".  Targets may link only to libraries.  CMake is dropping the item.
-- Generating done
-- Build files have been written to: /home/onzone/libopenshot/build
```
Any help, please?

**Comment by JacksonRG** [comment]
Have you build libopenshot-audio already?

**Comment by onzone** [comment]
Yes..I have build libopenshot-audio..And it built fine..

**Comment by ferdnyc** [comment]
Ubuntu 16.04 has entered what Ubuntu calls "Extended Security Maintenance" (as of April 2021), basically it's no longer supported except for paying customers with critical issues with the core components.

It's therefore no longer supported by OpenShot/libopenshot, sorry. (You could build on it, but you'd have to upgrade CMake and FFmpeg at a minimum. Possibly GCC as well.)

**Comment by ferdnyc** [comment]
Oh! And Python. We've stopped supporting Python 3.4 and 3.5, which are both now end-of-life.

**Comment by ferdnyc** [comment]
Ah, it looks like you _have_ upgraded those things — that's good. The issue you were actually seeing was a boneheaded mistake I made in the FFmpeg linking, but it's now been fixed by #693 and #694. Try the latest `develop` code, it should do the right thing.
